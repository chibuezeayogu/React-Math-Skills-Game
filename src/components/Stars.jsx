import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

const Stars = ({ numberOfStars }) => (
  <div className="col-5">
    {_.range(numberOfStars).map(i => <i key={i} className="fa fa-star" />)}
  </div>
);

Stars.propTypes = {
  numberOfStars: PropTypes.number,
};

export default Stars;
