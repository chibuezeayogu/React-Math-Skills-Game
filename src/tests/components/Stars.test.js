/* 
global jest 
expect 
*/
import React from 'react';
import { shallow } from 'enzyme';

import Stars from '../../components/Stars';

describe('Stars Component', () => {
  const setup = () => {
    const props = {
      numberOfStars: 8
    };

    return shallow(<Stars {...props}/>);
  };
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
