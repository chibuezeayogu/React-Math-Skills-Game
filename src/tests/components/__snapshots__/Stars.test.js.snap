// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars Component should render correctly 1`] = `
<div
  className="col-5"
>
  <i
    className="fa fa-star"
    key="0"
  />
  <i
    className="fa fa-star"
    key="1"
  />
  <i
    className="fa fa-star"
    key="2"
  />
  <i
    className="fa fa-star"
    key="3"
  />
  <i
    className="fa fa-star"
    key="4"
  />
  <i
    className="fa fa-star"
    key="5"
  />
  <i
    className="fa fa-star"
    key="6"
  />
  <i
    className="fa fa-star"
    key="7"
  />
</div>
`;
